applications/ConstitutiveLawsApplication/custom_utilities/autom_differentiation_tangent_utilities.cpp

```diff
@@ -0,0 +1,3528 @@
+  r_Ct(1, 5) = -cr_Ct119 * r_strain[5];
+  r_Ct(2, 0) = -cr_Ct86 * (cr_Ct121 * cr_Ct54 + cr_Ct122 * cr_Ct75 + cr_Ct87);
+  r_Ct(2, 1) = -cr_Ct86 * (cr_Ct121 * cr_Ct94 + cr_Ct122 * cr_Ct89 + cr_Ct87);
+  r_Ct(2, 2) = cr_Ct118 * (cr_Ct113 - cr_Ct116 * cr_Ct123 * (cr_Ct115 + cr_Ct49 + cr_Ct96) + cr_Ct117 * cr_Ct123 * cr_Ct97);
+  r_Ct(2, 3) = -cr_Ct124 * r_strain[3];
+  r_Ct(2, 4) = -cr_Ct124 * r_strain[4];
+  r_Ct(2, 5) = -cr_Ct124 * r_strain[5];
+  r_Ct(3, 0) = -cr_Ct129 * cr_Ct130;
+  r_Ct(3, 1) = -cr_Ct130 * cr_Ct132;
+  r_Ct(3, 2) = -cr_Ct130 * cr_Ct133;
+  r_Ct(3, 3) = cr_Ct85 * (cr_Ct134 - cr_Ct135 * cr_Ct32 - cr_Ct136 * cr_Ct32);
+  r_Ct(3, 4) = cr_Ct139;
+  r_Ct(3, 5) = cr_Ct140;
+  r_Ct(4, 0) = -cr_Ct129 * cr_Ct141;
+  r_Ct(4, 1) = -cr_Ct132 * cr_Ct141;
+  r_Ct(4, 2) = -cr_Ct133 * cr_Ct141;
+  r_Ct(4, 3) = cr_Ct139;
+  r_Ct(4, 4) = cr_Ct85 * (cr_Ct134 - cr_Ct135 * cr_Ct33 - cr_Ct136 * cr_Ct33);
+  r_Ct(4, 5) = cr_Ct142;
+  r_Ct(5, 0) = -cr_Ct129 * cr_Ct143;
+  r_Ct(5, 1) = -cr_Ct132 * cr_Ct143;
+  r_Ct(5, 2) = -cr_Ct133 * cr_Ct143;
+  r_Ct(5, 3) = cr_Ct140;
+  r_Ct(5, 4) = cr_Ct142;
+  r_Ct(5, 5) = cr_Ct85 * (cr_Ct134 - cr_Ct135 * cr_Ct34 - cr_Ct136 * cr_Ct34);
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void AutomaticDifferentiationTangentUtilities<ModifiedMohrCoulombYieldSurface<ModifiedMohrCoulombPlasticPotential<6>>, static_cast<SizeType>(SofteningType::Exponential)>::
+CalculateTangentTensorAutomDiffIsotropicDamage(
+  ConstitutiveLaw::Parameters rValues
+  )
+{
+  const auto &r_props = rValues.GetMaterialProperties();
+  const double Young = r_props[YOUNG_MODULUS];
+  const double nu = r_props[POISSON_RATIO];
+  const double Gf = r_props[FRACTURE_ENERGY];
+  const double phi = r_props[FRICTION_ANGLE] * Globals::Pi / 180.0;
+  const double characteristic_length = AdvancedConstitutiveLawUtilities<VoigtSize>::CalculateCharacteristicLength(rValues.GetElementGeometry());
+  auto &r_Ct = rValues.GetConstitutiveMatrix();
+  const auto &r_strain = rValues.GetStrainVector();
+
+  const bool has_symmetric_yield_stress = r_props.Has(YIELD_STRESS);
+  const double threshold_compression = has_symmetric_yield_stress ? r_props[YIELD_STRESS] : r_props[YIELD_STRESS_COMPRESSION];
+  const double threshold_tension = has_symmetric_yield_stress ? r_props[YIELD_STRESS] : r_props[YIELD_STRESS_TENSION];
+
+  const double cr_Ct0 = nu - 1.0;
+  const double cr_Ct1 = nu*r_strain[0];
+  const double cr_Ct2 = nu*r_strain[1];
+  const double cr_Ct3 = nu*r_strain[2];
+  const double cr_Ct4 = cr_Ct0*r_strain[0];
+  const double cr_Ct5 = cr_Ct0*r_strain[1];
+  const double cr_Ct6 = cr_Ct0*r_strain[2];
+  const double cr_Ct7 = 2.0*nu;
+  const double cr_Ct8 = 1.0/(cr_Ct7 - 1.0);
+  const double cr_Ct9 = std::abs(threshold_compression/threshold_tension);
+  const double cr_Ct10 = tan(0.5*phi + 0.78539816339744828);
```

Comment:
The std::tan and std::sin are missing

Reply:
This always happens... Maybe we can add something to the `AutomaticDifferentiationUtitlities` to automatically do it.